esmvaltool/interface_scripts/cmor_check.py

```diff
@@ -17,6 +15,36 @@
 
 
 class CMORCheck(object):
+    """
+    Class used to check the CMOR-compliance of the data.
+
+    It can also fix some minor errors and does some minor data
+    homogeneization:
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube:
+        Iris cube to check
+    var_info: variables_info.VariableInfo
+        Variable info to check
+    frequency: str
+        Expected frequency for the data
+    fail_on_error: bool
+        If true, CMORCheck stops on the first error. If false, it collects
+        all possible errors before stopping
+    automatic_fixes: bool
+        If True, CMORCheck will try to apply automatic fixes for any
+        detected error, if possible
+
+    Attributes
+    ----------
+    frequency: str
+        Expected frequency for the data
+    automatic_fixes: bool
+        If True, CMORCheck will try to apply automatic fixes for any
+        detected error, if possible
+
+    """
 
```

Comment:
Are we going to follow this standard for documenting the rest of the preprocessor?
@bouweandela @nielsdrost @valeriupredoi @axel-lauer

Reply:
I think we decided to use numpy standard at one of the coding workshops. The only changes I have made on this branch are to get rid of some prospector warnings

Reply:
Indeed we decided to use the numpy docstring standard in issue #50. This still needs to be updated in the documentation though.